components/logins/src/db.rs

```diff
@@ -201,6 +166,41 @@ fn sqlcipher_3_compat(conn: &Connection) -> Result<()> {
     Ok(())
 }
 
+/// Opens an existing database and fetches the salt.
+/// This method is used by iOS consumers as part as the migration plan to store
+/// the salt outside of the sqlite db headers.
+///
+/// Will return an error if the database does not exist.
+pub fn open_and_get_salt(path: impl AsRef<Path>, encryption_key: &str) -> Result<String> {
```

Comment:
These don't call &self and can be taken out of the impl block (since their static) to keep things cleaner on the Uniffi side

Reply:
(Yeah, the only reason anyone would want to use these methods is to work around some weirdness on iOS)